admin_ui/backend/api/tools.py

```
@@ -242,6 +243,238 @@ async def preview_email_template(request: EmailTemplatePreviewRequest):
+                    is_global=bool(getattr(d, "is_global", False)),
+                    requires_channel=bool(getattr(d, "requires_channel", False)),
+                    max_execution_time=int(getattr(d, "max_execution_time", 0) or 0),
+                    timeout_ms=_safe_jsonable(getattr(d, "timeout_ms", None)),
+                    output_variables=_safe_jsonable(getattr(d, "output_variables", [])),
+                    parameters=params_out,
+                    has_input_schema=bool(getattr(d, "input_schema", None)),
+                    source=source,
+                )
+            )
+        return ToolCatalogResponse(tools=out, source="local_fallback")
+    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)